### DISCUSSION TITLE
A problem with  FB_OUTPUT_BISTABLE_MQTT function

### DISCUSSION THREAD

**Discussion by SzwedaM** [MichielVanwelsenaere/HomeAutomation.CoDeSys3#134] [Q&A] [answered]
Hi all, 
I have used this function in my project. It was working with no issue before. Unfortunately, I lost working this function after updating to the latest version. I am not able to control the relay from HA to my PLC now. I can see relay status changes on HA only - it looks like I have one way communication (from PLC to HA). I have tried to setup this function again using the new library with no success. I have copied Home Assistant samples to eliminate typing mistakes - still same problem. 

What was changed in the last version? 

Thank you for suggestions.
Regards
Mariusz

**Comment by meijer3** [comment, marked as answer]
I guess, one thing you can double check all the MQTT paths.
It has changed in some version, I dont know which.
See my struggle https://github.com/MichielVanwelsenaere/HomeAutomation.CoDeSys3/issues/124, maybe it helps yours?

**Reply by SzwedaM** [reply]
Thank you for your answer. As you suggested, the MQTTSubscribeTopic is the same for all my bistable relays. The name of the path is losing the last 4 characters as on the attached print screen. But why??

![image](https://user-images.githubusercontent.com/88900591/214048095-d7748864-379b-448c-85cf-bbc4871167f6.png)

**Reply by meijer3** [reply]
I am actually cleaning all these topics in a new PR #131 
For now you can increase all these string lengths. 
![image](https://user-images.githubusercontent.com/45625341/214048700-0fe4c9a7-e6eb-4730-8d68-f8c961bb8a42.png)

**Reply by SzwedaM** [reply]
I have increased the string length to 60. It is working now. 
Thank you for your assistance.
